core/src/tile/tileManager.cpp

```diff
@@ -237,6 +238,9 @@ void TileManager::updateTileSet(TileSet& _tileSet, const ViewState& _view,
                 entry.task = _tileSet.source->createTask(visTileId);
                 enqueueTask(_tileSet, visTileId, _view);
                 m_tilesInProgress++;
+            } else if (!entry.needsLoading() && !entry.isReady()) {
+                // tile not generated yet!
+                m_tilesInGeneration++;
             }
```

Comment:
I'm not yet convinced that we should have both. How about `m_tilesInProgress` counter for all tiles that are `!entry.isReady() || (entry.tile->sourceGeneration() < generation)` ?